### PR TITLE
Log preview lines in job-runner

### PR DESCRIPTION
**Pull request by NathanFlurry**

### CODE REVIEW

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/346?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #324** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #314** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/314?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #310** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #311** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #312** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #313** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/313?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #315** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #316** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #326** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #325** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #329** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/329?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #330** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/330?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #331** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #332** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #333** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/333?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #334** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/334?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #335** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/335?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **PR #336** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/336?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                    * **PR #337** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/337?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                      * **PR #338** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/338?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                        * **PR #339** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/339?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                          * **PR #340** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/340?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                            * **PR #341** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/341?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                              * **PR #342** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/342?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                * **PR #343** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/343?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                  * **PR #344** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/344?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                    * **PR #345** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                                      * **PR #346** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/346?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                                                        * **PR #347** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/347?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/346?utm_source=stack-comment).

File: lib/job-runner/src/main.rs [hunk 1]
```diff
@@ -16,6 +16,10 @@ const MAX_LINE_BYTES: usize = 1024;
 /// Maximum number of bytes to buffer before dropping logs
 const MAX_BUFFER_BYTES: usize = 1024 * 1024;
 
+/// Maximum number of lines to print to stdout for easy debugging why programs crash from Nomad's
+/// output.
+const MAX_PREVIEW_LINES: usize = 128;
```
Comment (AngelOnFira) [inline lib/job-runner/src/main.rs:21]: I found this like a bit hard to read. I think this is more clear:

```suggestion
/// Maximum number of lines to print to stdout for debugging. This helps
/// identify the reasons for program crashes based from Nomad's output.
const MAX_PREVIEW_LINES: usize = 128;
```

File: lib/job-runner/src/main.rs [hunk 2]
```diff
@@ -150,6 +154,9 @@ fn ship_logs(
 		// Throttles error logs
 		let mut throttle_error = throttle::Throttle::new(1, Duration::from_secs(60));
 
+		// How many lines have been logged  as a preview, see `MAX_PREVIEW_LINES`
+		let mut preview_iine_count = 0;
```
Comment (AngelOnFira) [inline lib/job-runner/src/main.rs:158]:
```suggestion
		// How many lines have been logged as a preview, see `MAX_PREVIEW_LINES`
		let mut preview_iine_count = 0;
```

**Review by AngelOnFira — CHANGES_REQUESTED** [review]

**Comment by NathanFlurry** [comment]
@AngelOnFira _poke_

**Review by MasterPtato — APPROVED** [review]

**Review by AngelOnFira — APPROVED** [review]

**Pull request merged**